From e95abf7f5bb256c37091574d8748717b428f3a16 Mon Sep 17 00:00:00 2001
From: Adithya R <gh0strider.2k18.reborn@gmail.com>
Date: Sun, 27 Jan 2019 14:17:13 +0000
Subject: [PATCH 2/5] klte-common: Update power profile for Pie

* Match the changes made in LineageOS/android_frameworks_base@3d422c3

* Since Pie, AOSP is forcing us to define speeds and power values per CPU cluster.
  For single cluster chipsets like MSM8974, define cluster0 with only 4 cores
  to retain the existing behaviour.

Change-Id: Ida1edb9cc0335ffdb0ec6857da02b84a6a9d1633
---
 overlay/frameworks/base/core/res/res/xml/power_profile.xml | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/overlay/frameworks/base/core/res/res/xml/power_profile.xml b/overlay/frameworks/base/core/res/res/xml/power_profile.xml
index 29d9dfb..6e257d3 100644
--- a/overlay/frameworks/base/core/res/res/xml/power_profile.xml
+++ b/overlay/frameworks/base/core/res/res/xml/power_profile.xml
@@ -17,7 +17,8 @@
         <value>4.5</value>
         <value>4.5</value>
     </array>
-    <array name="cpu.speeds">
+    <item name="cpu.clusters.cores">4</item>
+    <array name="cpu.core_speeds.cluster0">
         <value>2457600</value>
         <value>2265600</value>
         <value>1958400</value>
@@ -35,7 +36,7 @@
         <value>300000</value>
     </array>
     <item name="cpu.idle">3.1</item>
-    <array name="cpu.active">
+    <array name="cpu.core_power.cluster0">
         <value>348</value>
         <value>313</value>
         <value>265</value>
@@ -60,4 +61,4 @@
         <value>.2</value>
         <value>2</value>
     </array>
-</device>
\ No newline at end of file
+</device>
-- 
2.17.1

